api.lisp

```diff
@@ -0,0 +1,77 @@
+;;;; -*- Mode: Lisp; indent-tabs-mode: nil -*-
+
+(in-package :split-sequence)
+
+(declaim (inline list-long-enough-p check-bounds check-tests))
+
+(defun list-long-enough-p (list length)
+  (or (zerop length)
+      (not (null (nthcdr (1- length) list)))))
+
+(defun check-bounds (sequence start end)
+  (progn
+    (check-type start unsigned-byte "a non-negative integer")
+    (check-type end (or null unsigned-byte) "a non-negative integer or NIL")
+    (typecase sequence
+      (list
+       (when end
+         (unless (list-long-enough-p sequence end)
+           (error "The list is too short: END was ~S but the list is ~S elements long."
+                  end (length sequence)))))
+      (t
+       (let ((length (length sequence)))
+         (unless end (setf end length))
+         (unless (<= start end length)
+           (error "Wrong sequence bounds. START: ~S END: ~S" start end)))))))
+
+(defun check-tests (test test-p test-not test-not-p)
+  (when (and test test-p test-not test-not-p)
+    (error "Cannot specify both TEST and TEST-NOT."))
```

Comment:
use http://www.lispworks.com/documentation/HyperSpec/Body/e_progra.htm

Reply:
Sure. I would have to define a custom condition type that is a subtype of `PROGRAM-ERROR` in order to use it here with a custom error message. `PROGRAM-ERROR` is not a `SIMPLE-CONDITION` on its own.